packages/core/saved-objects/core-saved-objects-migration-server-internal/src/zdt/model/stages/init.ts

```diff
@@ -60,37 +79,81 @@ export const init: ModelStage<
   // Index was found. This is the standard scenario, we check the model versions
   // compatibility before going further.
   const currentMappings = indices[currentIndex].mappings;
-  const versionCheck = checkVersionCompatibility({
-    mappings: currentMappings,
-    types,
-    source: 'mappingVersions',
-    deletedTypes: context.deletedTypes,
-  });
+
+  // Index is already present, so we check which algo was last used on it
+  const currentAlgo = checkIndexCurrentAlgorithm(currentMappings);
```

Comment:
Changes in the `init` stage start here. I added comments to describe the logic of the workflow.